src/character_functions.cpp

```diff
@@ -1142,12 +1142,22 @@ void find_ammo_helper( T &src, const item &obj, bool empty, Output out, bool nes
     } else {
         // find compatible magazines excluding those already loaded in tools/guns
         const auto mags = obj.magazine_compatible();
+        const std::set<ammotype> &ammo = obj.ammo_types();
 
-        src.visit_items( [&nested, &out, mags, empty]( item * node ) {
+        src.visit_items( [&nested, &out, mags, empty, ammo]( item * node ) {
```

Comment:
```suggestion
        src.visit_items( [&nested, &out, mags, empty, &ammo]( item * node ) {
```

capturing without referencing it `&` copies the value, which might slow down loop.

https://learn.microsoft.com/en-us/cpp/cpp/lambda-expressions-in-cpp?view=msvc-170#capture-clause